dashboard/app/models/script_level.rb

```diff
@@ -372,6 +372,48 @@ def summarize_as_bonus
     }.camelize_keys
   end
 
+  def summarize_for_teacher_panel(student)
+    contained_levels = levels.map(&:contained_levels).flatten
+    if contained_levels.any?
+      user_level = student.last_attempt_for_any(contained_levels)
+      contained = true
+    else
+      user_level = student.last_attempt_for_any(levels)
+      contained = false
+    end
+
+    status = activity_css_class(user_level)
+    passed = [SharedConstants::LEVEL_STATUS.passed, SharedConstants::LEVEL_STATUS.perfect].include?(status)
+
+    if user_level
+      paired = user_level.paired?
+      if UserLevel.most_recent_driver(@script, @level, student)
+        driver = UserLevel.most_recent_driver(@script, @level, student)[0] # the name is in array index 0
```

Comment:
`UserLevel.most_recent_driver` hits the database -- let's not call it twice. here is one way: `driver = UserLevel.most_recent_driver(@script, @level, student).try(:[], 0)` works because in ruby `array[0]` is just calling the `[]` method with argument `0`. could also do something less tricky like:
```
driver_info = UserLevel.most_recent_driver(...)
driver = driver_info[0] if driver_info
```